### PR TITLE
Remove the unnecessary DatasetProvider hierarchy

### PR DESCRIPTION
**Pull request by chtyim**
- After UsageRegistry moved out of DatasetService, we no longer need the DatasetProvider class hierarchy.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/module/lib/DatasetModules.java [hunk 1]
```diff
@@ -36,29 +36,33 @@ private DatasetModules() {}
...
-      throw new IllegalArgumentException(msg);
+      return (DatasetModule) clazz.newInstance();
     }
-    return module;
+    if (Dataset.class.isAssignableFrom(clazz)) {
+      return new SingleTypeModule(clazz);
+    }
+
+    String msg = String.format(
+      "Cannot use class %s to instantiate dataset module: it must be of type DatasetModule or Dataset",
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/module/lib/DatasetModules.java]: use semicolon instead of colon

Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/module/lib/DatasetModules.java]: It's from the old code above. Will have it changed.

**Comment by anwar6953** [comment]
1 minor comment; remainder LGTM

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT4207

**Comment by chtyim** [comment]
Addressed the comment about the semicolon. Squashed and merging.

**Pull request merged**